From 29067adfad3b76d228d38fd4b73e12f2e0864697 Mon Sep 17 00:00:00 2001
From: hyphop <art@khadas.com>
Date: Sun, 26 Apr 2020 01:23:04 +0900
Subject: [PATCH 30/53] add_fdtcontroladdr_end_to_env

addition fdtcontroladdr_end variable
same as fdtcontroladdr but for end of fdt data
can use for embed logo or other data
---
 common/board_r.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/common/board_r.c b/common/board_r.c
index 66760288..097682e5 100644
--- a/common/board_r.c
+++ b/common/board_r.c
@@ -481,8 +481,11 @@ static int initr_env(void)
 	else
 		env_set_default(NULL, 0);
 #ifdef CONFIG_OF_CONTROL
-	env_set_hex("fdtcontroladdr",
-		    (unsigned long)map_to_sysmem(gd->fdt_blob));
+	ulong fdt_begin = (unsigned long)map_to_sysmem(gd->fdt_blob);
+	env_set_hex("fdtcontroladdr", fdt_begin );
+	ulong fdt_size = (unsigned long)fdt_totalsize(gd->fdt_blob);
+//	printf("FDT SIZE: %lu\n", fdt_size);
+	env_set_hex("fdtcontroladdr_end", fdt_begin + fdt_size );
 #endif
 
 	/* Initialize from environment */
-- 
2.17.1

